### DISCUSSION TITLE
TMF URI ressources changes impact on PanIdQuery for MTD

### DISCUSSION THREAD

**Discussion by EricSTMicro** [openthread/openthread#8918] [Q&A] [answered]
Hello,

after updating our OpenThread branch from  a commit from last July to a more recent one (5th February 2023), I have seen a behavior change when using the function `OtCommissionerPanIdQuery()` from a FTD device sending it to a MTD device.

[This commit](https://github.com/openthread/openthread/commit/812c36f6581967b3d9af8361f00716c75f47473c) from 7th October 2022 introduced a new way of handling the URI ressources:
```
bool Agent::HandleResource(const char *aUriPath, Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
{
    bool didHandle = true;
    Uri  uri       = UriFromPath(aUriPath);

#define Case(kUri, Type)                                     \
    case kUri:                                               \
        Get<Type>().HandleTmf<kUri>(aMessage, aMessageInfo); \
        break

    switch (uri)
    {
        Case(kUriAddressError, AddressResolver);
        Case(kUriEnergyScan, EnergyScanServer);
        Case(kUriActiveGet, MeshCoP::ActiveDatasetManager);
        Case(kUriPendingGet, MeshCoP::PendingDatasetManager);

#if OPENTHREAD_FTD
        Case(kUriAddressQuery, AddressResolver);
        Case(kUriAddressNotify, AddressResolver);
        Case(kUriAddressSolicit, Mle::MleRouter);
        Case(kUriAddressRelease, Mle::MleRouter);
        Case(kUriActiveSet, MeshCoP::ActiveDatasetManager);
        Case(kUriPendingSet, MeshCoP::PendingDatasetManager);
        Case(kUriLeaderPetition, MeshCoP::Leader);
        Case(kUriLeaderKeepAlive, MeshCoP::Leader);
        Case(kUriServerData, NetworkData::Leader);
        Case(kUriCommissionerGet, NetworkData::Leader);
        Case(kUriCommissionerSet, NetworkData::Leader);
        Case(kUriAnnounceBegin, AnnounceBeginServer);
        Case(kUriPanIdQuery, PanIdQueryServer);     <----------- only FTD can access this case
        Case(kUriRelayTx, MeshCoP::JoinerRouter);
#endif

[...] // Some more code after

}
```
So before that the MTD was able to handle the PanidQuery, now it isn't.

**My question is: is this a mistake or is it an intended limitation?**

Thank you for reading me,

Eric

**Comment by abtink** [comment, marked as answer]
Thanks @EricSTMicro 

No. It was not an intentional change.
I submitted https://github.com/openthread/openthread/pull/8919 to make it available on MTD (along with `AnnounceBeginServer`)

Though practically I am not sure Pan ID Query on MTD (specially on sleepy) would be very useful.

**Reply by EricSTMicro** [reply]
Hello Abtink, thank you for your quick answer.
In fact we have automated testing and this one is failing for MTD since last update. If you told me it was a removed feature, I'll have simply removed it from our testing :)
